server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesUpdateProcessor.java

```diff
@@ -134,18 +143,38 @@ private void update(MessageId messageId, UpdateMessagePatch updateMessagePatch,
         }
     }
 
-    private void setInMailboxes(MessageId messageId, UpdateMessagePatch updateMessagePatch, MailboxSession mailboxSession) throws MailboxException {
+    private void setInMailboxes(MessageId messageId, UpdateMessagePatch updateMessagePatch, Stream<Flags> originalFlags, MailboxSession mailboxSession) throws MailboxException {
         Optional<List<String>> serializedMailboxIds = updateMessagePatch.getMailboxIds();
         if (serializedMailboxIds.isPresent()) {
             List<MailboxId> mailboxIds = serializedMailboxIds.get()
                 .stream()
                 .map(mailboxIdFactory::fromString)
                 .collect(Guavate.toImmutableList());
 
+            assertNoDraftMailbox(mailboxSession, mailboxIds);
```

Comment:
preventDraftMessageInNonDraftsMailbox ?

Reply:
Yet another naming sheme? After **assert** where we have no validation results, after **validate** where we have validation results, what makes us want using the naming "prevent"?